src/lib.rs

```diff
@@ -0,0 +1,230 @@
+extern crate ini;
+extern crate regex;
+extern crate semver;
+
+use ini::Ini;
+use regex::Regex;
+use std::default::Default;
+use std::fmt::{self, Display, Formatter};
+use std::path::Path;
+use std::str::FromStr;
+
+#[derive(Default)]
+pub struct AppVersion {
+    pub build_id: Option<String>,
+    pub code_name: Option<String>,
+    pub version_string: Option<String>,
+    pub source_repository: Option<String>,
+    pub source_stamp: Option<String>
```

Comment:
There should be trailing commas on all lists of this sort. More cases of this below, but I won’t add comments for them.